metal/datasource_metal_facility.go

```diff
@@ -33,6 +33,26 @@ func dataSourceMetalFacility() *schema.Resource {
 				Elem:        &schema.Schema{Type: schema.TypeString},
 				Computed:    true,
 			},
+			"capacity": {
+				Type:        schema.TypeList,
+				Description: "Optional capacity specification",
+				MaxItems:    1,
```

Comment:
Should we allow for multiple capacity blocks?  What if I want a facility with 2 n2 devices and 3 c3 devices.

Reply:
I suppose we could think about that later and users could just use multiple `data "metal_facility"` blocks for now. (since this resource conditionally represents a `code` predefined facility)

Reply:
It's actually not too hard to do it right now, the response on a capacity request looks following:

```
{"servers":[
  {"facility":"sv15","plan":"c3.small.x86","quantity":1,"available":true},
  {"facility":"sv15","plan":"c3.medium.x86","quantity":3,"available":false}
]}
```

We can agree that the datasource will throw an error if any of the sections has `available:false`. The error will actually say which capacity requirement can't be satisfied.